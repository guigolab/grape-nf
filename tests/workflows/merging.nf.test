nextflow_workflow {

    name "Test Workflow merging"
    script "workflows/merging.nf"
    workflow "merging"

    tag "workflow"

    test("Should merge genome BAM files using sambamba") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
                mergeBamTool = "sambamba"
                markdupTool = "sambamba"
            }
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample2", "test2", file("${baseDir}/data/test2/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true],
                    [ "sample2", "test3", file("${baseDir}/data/test3/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true]
                ])
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Should merge genome BAM files using samtools") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
                mergeBamTool = "samtools"
                markdupTool = "sambamba"
            }
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample2", "test2", file("${baseDir}/data/test2/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true],
                    [ "sample2", "test3", file("${baseDir}/data/test3/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true]
                ])
                input[1] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Should merge all BAM files using samtools") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
                mergeBamTool = "samtools"
                markdupTool = "sambamba"
            }
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample2", "test2", file("${baseDir}/data/test2/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true],
                    [ "sample2", "test3", file("${baseDir}/data/test3/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true]
                ])
                input[1] = Channel.from([
                    [ "sample2", "test2", file("${baseDir}/data/test2/sample2_m4_n10_toTranscriptome_sorted.bam"), "bam", "TranscriptomeAlignments", true],
                    [ "sample2", "test3", file("${baseDir}/data/test3/sample2_m4_n10_toTranscriptome_sorted.bam"), "bam", "TranscriptomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Should merge all BAM files using sambamba") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
                mergeBamTool = "sambamba"
                markdupTool = "sambamba"
            }
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample2", "test2", file("${baseDir}/data/test2/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true],
                    [ "sample2", "test3", file("${baseDir}/data/test3/sample2_m4_n10_toGenome_sorted.bam"), "bam", "GenomeAlignments", true]
                ])
                input[1] = Channel.from([
                    [ "sample2", "test2", file("${baseDir}/data/test2/sample2_m4_n10_toTranscriptome_sorted.bam"), "bam", "TranscriptomeAlignments", true],
                    [ "sample2", "test3", file("${baseDir}/data/test3/sample2_m4_n10_toTranscriptome_sorted.bam"), "bam", "TranscriptomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
