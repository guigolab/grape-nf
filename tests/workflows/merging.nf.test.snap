{
    "Should merge genome BAM files using sambamba": {
        "content": [
            {
                "0": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,92d1c59a6e5b111f7a277135310415c2",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    
                ],
                "genomeAlignments": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,92d1c59a6e5b111f7a277135310415c2",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "transcriptomeAlignments": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T15:07:56.427844"
    },
    "Should merge genome BAM files using samtools": {
        "content": [
            {
                "0": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,9c721443896d97df3f2151ce5397c512",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    
                ],
                "genomeAlignments": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,9c721443896d97df3f2151ce5397c512",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "transcriptomeAlignments": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T15:08:05.780447"
    },
    "Should merge all BAM files using sambamba": {
        "content": [
            {
                "0": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,92d1c59a6e5b111f7a277135310415c2",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toTranscriptome.bam:md5,933baf1446265d924a5bc02a1f75158c",
                        "bam",
                        "TranscriptomeAlignments",
                        true
                    ]
                ],
                "genomeAlignments": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,92d1c59a6e5b111f7a277135310415c2",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "transcriptomeAlignments": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toTranscriptome.bam:md5,933baf1446265d924a5bc02a1f75158c",
                        "bam",
                        "TranscriptomeAlignments",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T15:08:27.346896"
    },
    "Should merge all BAM files using samtools": {
        "content": [
            {
                "0": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,9c721443896d97df3f2151ce5397c512",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toTranscriptome.bam:md5,2b2f97df2a0f72d280834ced7e046ed3",
                        "bam",
                        "TranscriptomeAlignments",
                        true
                    ]
                ],
                "genomeAlignments": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,9c721443896d97df3f2151ce5397c512",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "transcriptomeAlignments": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toTranscriptome.bam:md5,2b2f97df2a0f72d280834ced7e046ed3",
                        "bam",
                        "TranscriptomeAlignments",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T15:08:16.222547"
    }
}